### ISSUE TITLE
Login failed - Password Incorrect

### ISSUE THREAD

**Issue by christianrj** [tuaris/freebsd-virtualmin#1]
I Installed this today on a fresh FreeBSD 10.2. Everything seems to be installed, but I can't login to webmin.

It always returns an error that the password is incorrect. I'm using the root user to login.

But I can login just fine via ssh.

How to fix this?

**Comment by tuaris** [comment]
I don't believe Webmin is using PAM for authentication in FreeBSD.   It's using a non-integrated account.  You should be able to sign on using the 'admin' username and the root password.

**Comment by christianrj** [comment]
Not working.

With user admin and root password:

"Login failed. Please try again."

It's possible to change the password for webmin admin user via command line in FreeBSD?

**Comment by tuaris** [comment]
Try

```
/usr/local/libexec/webmin/changepass.pl /usr/local/etc/webmin admin <newpass>
```

**Comment by christianrj** [comment]
```
/usr/local/libexec/webmin/changepass.pl: Command not found.
```

But with the help of find command:

```
find / -name "changepass.pl"
/usr/local/lib/webmin/changepass.pl
```

So the command that worked was:

```
/usr/local/lib/webmin/changepass.pl /usr/local/etc/webmin admin <newpass>
```

I've managed to login, and the post configuration started, but an error occurred, stating that MySQL is not installed:

```
MySQL cannot be enabled, as it is not installed on your system
```

![mysql](https://cloud.githubusercontent.com/assets/8201830/12488671/da624c80-c053-11e5-88c2-f4e5e3a6c4b2.jpg)

**Comment by tuaris** [comment]
I haven't gotten to setting up MySQL in the install scripts, but it should be in the package repositories.  If you want to try and see if it works manually:

```
pkg install -y mysql56-server
```

That **should** work.  If not, maybe I'll re-visit this and do some additional work on it today.

**Comment by tuaris** [comment]
I tried to reproduce your password issue without any luck.  Was the username of an initial user you setup during FreeBSD's install 'admin'? If so, was admin placed into the 'wheel' secondary group?

**Comment by christianrj** [comment]
No. Only user configured in FreeBSD install was root. Not a single user was created.

Also, on virtualmin post setup, it complained about the mysql password. I fixed it manually running webmin mysql module to enter the password, and after running the post setup again, this error occured:
![mysql2](https://cloud.githubusercontent.com/assets/8201830/12489864/a84a9a6c-c059-11e5-9003-26f436059d67.JPG)

So, as suggestions, after the finish of setup script, add an option for the user to set the admin password for webmin/virtualmin. And, add MySQL to the setup script so it can be used and the post config can be completed. At least MySQL shoud be installed. Postgree is not important, but MySQL is.

**Comment by tuaris** [comment]
Thanks, I will  try and debug.

A known issue: MySQL setup throws an error when you select anything other than "Leave Default Settings".

**Comment by christianrj** [comment]
I skipped the MySQL part at least to finish the post configuration. And it seems to be everything working. Thank you for all your work!!

But without MySQL this setup is almost non functional. We need to get MySQL to setup with the system automatically like all the other packages. 

Now, I will test this on a remote system with real valid IPs, and test the following:

Server 1: main web server and primary DNS server
Server 2: Secondary DNS server (controlled by server 1)

Thank you!!

**Comment by tuaris** [comment]
MySQL has been added to the install script:
19e16cb5eb819d6d50ad8e08051d9c8be4a71a8f
Still deciding what to do about the password issue.

**Comment by christianrj** [comment]
Why not ask at the end of the install script to create the admin user?
